File: backend/src/main/java/com/smithlife/backend/entity/Reservation.java
package com.smithlife.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "reservation")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "reservation_id")
    private Long reservationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "slot_id", nullable = false)
    private TimeSlot slot;

    @Column(name = "reservation_no", nullable = false, unique = true, length = 20)
    private String reservationNo;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private Status status;

    @Column(name = "reserved_at", nullable = false)
    private LocalDateTime reservedAt;

    @Column(name = "cancelled_at")
    private LocalDateTime cancelledAt;

    @OneToMany(mappedBy = "reservation", fetch = FetchType.LAZY)
    private List<UsageHistory> usageHistories;

    public enum Status {
        CONFIRMED, CANCELLED, COMPLETED, NO_SHOW
    }
}

File: frontend/lib/data/models/reservation/reservation.dart
import 'time_slot.dart';

class Reservation {
  final int reservationId;
  final String reservationNo;
  final String status;
  final String reservedAt;
  final String? cancelledAt;
  final TimeSlot slot;
  final String? userName;

  Reservation({
    required this.reservationId,
    required this.reservationNo,
    required this.status,
    required this.reservedAt,
    this.cancelledAt,
    required this.slot,
    this.userName,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) {
    return Reservation(
      reservationId: json['reservationId'],
      reservationNo: json['reservationNo'],
      status: json['status'],
      reservedAt: json['reservedAt'],
      cancelledAt: json['cancelledAt'],
      slot: TimeSlot.fromJson(json['slot']),
      userName: json['userName'],
    );
  }
}
